workflow-runtime/src/commonTest/kotlin/com/squareup/workflow1/internal/WorkStealingDispatcherTest.kt

```diff
@@ -0,0 +1,439 @@
+    val dispatcher = WorkStealingDispatcher(testDispatcher)
+
+    expect(0)
+    dispatcher.dispatch {
+      expect(2)
+      dispatcher.dispatch {
+        expect(4)
+      }
+      expect(3)
+    }
+    expect(1)
+
+    dispatcher.advanceUntilIdle()
+    expect(5)
+  }
+
+  @Test fun advanceUntilIdle_canBeCalledReentrantly() {
+    val testDispatcher = StandardTestDispatcher()
+    val dispatcher = WorkStealingDispatcher(testDispatcher)
+
+    expect(0)
+    dispatcher.dispatch {
+      expect(2)
+      dispatcher.dispatch {
+        expect(4)
+      }
+      expect(3)
+
+      dispatcher.advanceUntilIdle()
+
+      expect(5)
+      dispatcher.dispatch {
+        expect(7)
+      }
+      expect(6)
+    }
+    expect(1)
+
+    dispatcher.advanceUntilIdle()
+    expect(8)
+  }
+
+  @Test fun advanceUntilIdle_noopsWhenNoTasks() {
+    val dispatcher = WorkStealingDispatcher(StandardTestDispatcher())
+
+    // Just make sure this doesn't throw when the queue is empty.
+    dispatcher.advanceUntilIdle()
+  }
+
+  @OptIn(ExperimentalCoroutinesApi::class)
+  @Test
+  fun limitedParallelism_unsupported() {
+    val dispatcher = WorkStealingDispatcher(NoopContinuationInterceptor())
+
+    assertFailsWith<UnsupportedOperationException> {
+      dispatcher.limitedParallelism(2)
+    }
+  }
+
+  @Test fun integration_unconfined() = runTest {
+    val dispatcher = WorkStealingDispatcher(Dispatchers.Unconfined)
+
+    expect(0)
+    launch(dispatcher) {
+      expect(1)
+    }
+    expect(2)
+  }
+
+  @Test fun integration_confined_whenAdvanceUntilIdle() = runTest {
+    val testDispatcher = StandardTestDispatcher()
+    val dispatcher = WorkStealingDispatcher(testDispatcher)
+
+    expect(0)
+    launch(dispatcher) {
+      expect(2)
+    }
+    expect(1)
+
+    dispatcher.advanceUntilIdle()
+    expect(3)
+  }
+
+  @Test fun integration_confined_whenDispatched() = runTest {
+    val testDispatcher = StandardTestDispatcher()
+    val dispatcher = WorkStealingDispatcher(testDispatcher)
+
+    expect(0)
+    launch(dispatcher) {
+      expect(2)
+    }
+    expect(1)
+
+    testDispatcher.scheduler.advanceUntilIdle()
+    expect(3)
+  }
+
+  @Test fun integration_yield_whenAdvanceUntilIdle() = runTest {
+    val testDispatcher = StandardTestDispatcher()
+    val dispatcher = WorkStealingDispatcher(testDispatcher)
+
+    launch(dispatcher) {
+      expect(0)
+      yield()
+      expect(2)
+    }
+    launch(dispatcher) {
+      expect(1)
+      yield()
+      expect(3)
+    }
+
+    dispatcher.advanceUntilIdle()
+    expect(4)
+  }
+
+  @Test fun integration_yield_whenDispatched() = runTest {
+    val testDispatcher = StandardTestDispatcher()
+    val dispatcher = WorkStealingDispatcher(testDispatcher)
+
+    launch(dispatcher) {
+      expect(0)
+      yield()
+      expect(2)
+    }
+    launch(dispatcher) {
+      expect(1)
+      yield()
+      expect(3)
+    }
+
+    testDispatcher.scheduler.advanceUntilIdle()
+    expect(4)
+  }
+
```

Comment:
No `delay()` tests here. But again, what do we want do with that? if we're going to ignore, then we need to make that clear.